lib/topological_inventory/orchestrator/object_database.rb

```diff
@@ -0,0 +1,24 @@
+module TopologicalInventory
+  module Orchestrator
+    class ObjectDatabase
```

Comment:
This whole thing feels overkill to me, as what you have built is basically a specialized Hash, where the key is a one-line transformation.  That is, instead of db.add earlier, I could just see `db = {}; db[key_from_object(object)] = object` (more or less)